### PR TITLE
OKTA-258188: Add Group Rules API to Postman Collections

### PR DESCRIPTION
**Pull request by greyyin-okta**
## Description:
- **What's changed?** 
-- Add Group Rule Operation APIs
-- Divide all group APIs into 4 folders, i.e., Group Operations, Group Member Operations, Group Rule Operations, and Group Rule Operations
-- Fix mistakes on the API reference page
-- Update the Postman URL for group APIs

- **Is this PR related to a Monolith release?** 
-- No
### Resolves:

* [OKTA-258188](https://oktainc.atlassian.net/browse/OKTA-258188)

@okta/eng-extensibility

### CODE REVIEW

**Comment by jakubvul-okta** [comment]
Hi Grey, normally our Postman collections are 1:1 with our Reference pages. From what I can tell, Group Rules are documented under the Groups API: https://developer.okta.com/docs/reference/api/groups/#group-rule-operations

Is there a reason this is a separate Postman collection, rather than just an addition to the Groups Postman collection?

**Comment by greyyin-okta** [comment]
@jakubvul Good point Jakub, I will move the `Group Rule` into `Group`.

File: packages/@okta/vuepress-site/.vuepress/public/docs/api/postman/groups.json [hunk 1]
```diff
@@ -1,296 +1,1020 @@
...
+								"lifecycle",
+								"activate"
+							]
+						}
+					},
+					"response": []
+				}
+			],
+			"protocolProfileBehavior": {}
 		}
```
Comment (greyyin-okta) [inline packages/@okta/vuepress-site/.vuepress/public/docs/api/postman/groups.json]: Only this part is newly added. All the rest APIs stay the same, but are just organized into different folders.

File: packages/@okta/vuepress-site/docs/reference/api/groups/index.md [hunk 2]
```diff
@@ -1194,13 +1195,13 @@ Updates a group rule.
...
 | condition.expression.type           | currently it is : urn:okta:expression:1.0                                                               | Body      | String                            | TRUE     |         |
 | condition.people.users.exclude      | userIds which would be excluded when rules are processed                                                | Body      | String                            | FALSE    |         |
 | condition.people.groups.exclude     | is currently not supported                                                                              | Body      | String                            | FALSE    |         |
-| actions.assignUserToGroups.groupIds | List of groupIds to which users would be added.							        | Body      | String                            | TRUE     |         |
+| actions.assignUserToGroups.groupIds | List of groupIds to which users would be added.                                                         | Body      | String                            | TRUE     |         |
```
[the remark this answers is not in the document]
Comment (greyyin-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: I just formatted this line, and didn't change the content. Should I change `List` to `Array`?

[the remark this answers is not in the document]
Comment (jakubvul-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: Please.

File: packages/@okta/vuepress-site/docs/reference/api/groups/index.md [hunk 3]
```diff
@@ -1300,6 +1301,8 @@ Lists all group rules for your organization.
 | -------------- | ------------------------------------------------------------ | ---------- | --------------------------------- | -------- | ------- |
 | limit          | Specifies the number of rule results in a page               | Query      | Number                            | FALSE    | 50      |
 | after          | Specifies the pagination cursor for the next page of rules   | Query      | String                            | FALSE    |         |
+| search         | Specifies the keyword of rules to search for                 | Query      | String                            | FALSE    |         |
+| expand         | If specified as groupIdToGroupNameMap, then show group names | Query      | String                            | FALSE    |         |
```
[the remark this answers is not in the document]
Comment (greyyin-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: Should be these four, i.e., `limit`, `after`, `search`, and `expand`.

[the remark this answers is not in the document]
Comment (jakubvul-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: I meant any other possible values for the `expand` param.

[the remark this answers is not in the document]
Comment (greyyin-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: Although it seems weird, but `groupIdToGroupNameMap` is hard-coded in the code.

File: packages/@okta/vuepress-site/.vuepress/public/docs/api/postman/groups.json [hunk 4]
```diff
@@ -1,296 +1,1020 @@
...
-			"time": 1448741359443,
-			"name": "List Groups with Profile Updated after Timestamp",
-			"description": "",
-			"collectionId": "05649232-8eeb-7fe6-8764-be67a95429ea",
-			"responses": []
-		},
+	"info": {
+		"_postman_id": "7ede3714-5721-4bc3-b528-36ecfd22fc72",
+		"name": "Groups (Okta API)",
+		"description": "The [Okta Groups API](/docs/api/rest/groups.html) provides operations to manage your organization groups and their user members.",
```
[the remark this answers is not in the document]
Comment (greyyin-okta) [reply packages/@okta/vuepress-site/.vuepress/public/docs/api/postman/groups.json]: Good catch!

File: packages/@okta/vuepress-site/docs/reference/api/groups/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakubvul-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: Then I would just ask that you make this (and the one on line 1421) code formatted.

[the remark this answers is not in the document]
Comment (greyyin-okta) [reply packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: Done!

**Comment by greyyin-okta** [comment]
Thanks for the review @jakubvul 🚀

File: packages/@okta/vuepress-site/docs/reference/api/groups/index.md [hunk 5]
```diff
@@ -1093,11 +1093,12 @@ Creates a group rule to dynamically add users to the specified group if they mat
...
 | condition.expression.type           | currently it is : urn:okta:expression:1.0                                                               | Body      | String                            | TRUE     |         |
 | condition.people.users.exclude      | userIds which would be excluded when rules are processed                                                | Body      | String                            | FALSE    |         |
 | condition.people.groups.exclude     | is currently not supported                                                                              | Body      | String                            | FALSE    |         |
-| actions.assignUserToGroups.groupIds | List of groupIds to which users would be added.							        | Body      | String                            | TRUE     |         |
+| actions.assignUserToGroups.groupIds | List of groupIds to which users would be added.                                                         | Body      | String                            | TRUE     |         |
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/groups/index.md]: This should also be "Array".

**Comment by greyyin-okta** [comment]
@alexandrazhang-okta has gone through this PR, and looks good to her. 🚀

**Pull request merged**